fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -653,17 +680,42 @@ void scanWithSynonymIndex() {
             // "hullo" is synonym of "hello"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(1, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"record\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "record", "layer"))), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
             // it doesn't match due to the "recor"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(0, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"recor\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "recor", "layer"))), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
             // "hullo" is synonym of "hello", and "show" is synonym of "record"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(1, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"show\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "show", "layer"))), null, ScanProperties.FORWARD_SCAN)
+                    .getCount().join());
+        }
+    }
+
+    @Test
+    void scanWithCombinedSetsSynonymIndex() throws IOException {
+        // The COMBINED_SYNONYM_SETS adds this extra line to our synonym set:
+        // 'synonym', 'nonsynonym'
+        try (InputStream is1 = new USEnglishSynonymMapConfig().getSynonymInputStream();
+             InputStream is2 = SynonymMapConfig.openFile("test.txt");
+             InputStream is = new SequenceInputStream(is1, is2)) {
+            SynonymMapRegistryImpl.instance().register(COMBINED_SYNONYM_SETS, is);
```

Comment:
Rather than call `register` explicitly here, would it make more sense to use the service loader, adding a new `@AutoService(SynonymMapConfig.class)` nested in the test that adds these so they are always there. This would be more in line with the way `IndexMaintainerFactory` works.

This also allows us to make `SynonymMapRegistry` immutable (removing both the `register` and `reset` methods)

Reply:
I am not sure why this allows us to remove the reset method?